cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,264 @@
+
+Please see https://azure.github.io/azure-workload-identity/docs/concepts.html for more information on how the AzWI mutating admission webhook works.
+
+### Architecture Diagram
+
+![Architecture Diagram](./2024-04-azure-workload-identity/architecture.png)
+
+### Detailed Design
+
+#### Advantages (of each option considered)
+
+n/a
+
+#### Disadvantages (of each option considered)
+
+n/a
+
+#### Proposed Option
+
+n/a
+
+### API design
+
+```diff
+enum AzureCredentialKind {
+  @doc("The Service Principal Credential")
+  ServicePrincipal,
++
++  @doc("The Workload Identity Credential")
++  WorkloadIdentity,
+}
+```
+
+```diff
++@doc("The properties of Azure Workload Identity credential storage")
++model AzureWorkloadIdentityProperties extends AzureCredentialProperties {
++  @doc("Workload Identity kind")
++  kind: AzureCredentialKind.WorkloadIdentity;
++
++  @doc("clientId for WorkloadIdentity")
++  clientId: string;
++
++  @doc("tenantId for WorkloadIdentity")
++  tenantId: string;
++
++  @doc("The storage properties")
++  storage: CredentialStorageProperties;
++}
+```
+
+### CLI Design
+
+`rad init --full` will need to be updated to ask the user to choose between service principal auth and Azure Workload Identity.
+
+We will need to update the `rad credential register azure` command to include two subcommands - `sp` and `wi`. The existing logic for `rad credential register azure` will be moved to `rad credential register azure sp`. We will add a new `rad credential register azure wi` command to allow the user to register the Entra ID Application Client ID for Azure Workload Identity.
+
+We will need to update the Radius Helm chart to allow the user to enable Azure Workload Identity during installation with the `global.azureWorkloadIdentity.enabled` value. We will need to update the installation flow (for `rad init` and `rad install kubernetes`) to annotate the Radius control plane pods with the `azure.workload.identity/use: "true"` label if the user sets this value in the Helm chart.
+
+We will need to update the `rad credential show` command to show the Entra ID Application Client ID for Azure Workload Identity.
+
+### Implementation Details
+
+#### UCP
+
+UCP should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
+
+#### Bicep
+
+n/a
+
+#### Deployment Engine
+
+Deployment Engine should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
+
+#### Core RP
+
+Core RP should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
```

Comment:
> It should then authenticate with Azure using this Client ID and the signed service account token.

Do we expose any function to fetch this token?

Reply:
We don't do it manually, but yes it is possible to fetch the token. We can discuss with the work related to private registries.